### ISSUE TITLE
classy_vision resnext101_32x4d gives incorrect result when passed channel last Tensor

### ISSUE THREAD

**Issue by cpuhrsch** [pytorch/pytorch#62594]
## To Reproduce

Script
```
import torch
from classy_vision.models import build_model

torch.manual_seed(123)
dtype = torch.float16
model = build_model({"name": "resnext101_32x4d"}).eval().cuda()

# By default BatchNorm weights are set to 0. This prevents that.
model._initialize_weights(False)
model = model.to(dtype)
device = torch.device('cuda')
inp = torch.randn(2, 3, 50, 50, device=torch.device('cuda'), dtype=dtype)
out1 = model(inp)
inp = inp.contiguous(memory_format=torch.channels_last)
out2 = model(inp)

print((out1 - out2).abs().mean().item())

# Using float16 tolerances from torch/testing/_core.py
print(torch.allclose(out1, out2, rtol=1e-3, atol=1e-3))
```

Script output
```
0.0718994140625
False
```

## Expected behavior

```
$ python collect_env.py
Collecting environment information...
PyTorch version: N/A
Is debug build: N/A
CUDA used to build PyTorch: N/A
ROCM used to build PyTorch: N/A

OS: Ubuntu 18.04.5 LTS (x86_64)
GCC version: (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0
Clang version: 6.0.0-1ubuntu2 (tags/RELEASE_600/final)
CMake version: version 3.18.4
Libc version: glibc-2.17

Python version: 3.7.10 (default, Jun  4 2021, 14:48:32)  [GCC 7.5.0] (64-bit runtime)
Python platform: Linux-5.4.0-1041-aws-x86_64-with-debian-buster-sid
Is CUDA available: N/A
CUDA runtime version: 11.0.221
GPU models and configuration: Could not collect
Nvidia driver version: Could not collect
cuDNN version: Probably one of the following:
/usr/local/cuda-10.1/targets/x86_64-linux/lib/libcudnn.so.7.6.5
/usr/local/cuda-10.2/targets/x86_64-linux/lib/libcudnn.so.7.6.5
/usr/local/cuda-11.0/targets/x86_64-linux/lib/libcudnn.so.8.0.5
/usr/local/cuda-11.0/targets/x86_64-linux/lib/libcudnn_adv_infer.so.8.0.5
/usr/local/cuda-11.0/targets/x86_64-linux/lib/libcudnn_adv_train.so.8.0.5
/usr/local/cuda-11.0/targets/x86_64-linux/lib/libcudnn_cnn_infer.so.8.0.5
/usr/local/cuda-11.0/targets/x86_64-linux/lib/libcudnn_cnn_train.so.8.0.5
/usr/local/cuda-11.0/targets/x86_64-linux/lib/libcudnn_ops_infer.so.8.0.5
/usr/local/cuda-11.0/targets/x86_64-linux/lib/libcudnn_ops_train.so.8.0.5
/usr/local/cuda-11.1/targets/x86_64-linux/lib/libcudnn.so.8.0.5
/usr/local/cuda-11.1/targets/x86_64-linux/lib/libcudnn_adv_infer.so.8.0.5
/usr/local/cuda-11.1/targets/x86_64-linux/lib/libcudnn_adv_train.so.8.0.5
/usr/local/cuda-11.1/targets/x86_64-linux/lib/libcudnn_cnn_infer.so.8.0.5
/usr/local/cuda-11.1/targets/x86_64-linux/lib/libcudnn_cnn_train.so.8.0.5
/usr/local/cuda-11.1/targets/x86_64-linux/lib/libcudnn_ops_infer.so.8.0.5
/usr/local/cuda-11.1/targets/x86_64-linux/lib/libcudnn_ops_train.so.8.0.5
HIP runtime version: N/A
MIOpen runtime version: N/A

Versions of relevant libraries:
[pip3] numpy==1.20.3
[pip3] torch==1.10.0.dev20210729
[pip3] torch2trt==0.2.0
[pip3] torchaudio==0.10.0a0+2475e25
[pip3] torchvision==0.11.0.dev20210729
[conda] blas                      1.0                         mkl
[conda] cudatoolkit               11.1.74              h6bb024c_0    nvidia
[conda] mkl                       2021.3.0           h06a4308_520
[conda] mkl-service               2.4.0            py37h7f8727e_0
[conda] mkl_fft                   1.3.0            py37h42c9631_2
[conda] mkl_random                1.2.2            py37h51133e4_0
[conda] numpy                     1.20.3           py37hf144106_0
[conda] numpy-base                1.20.3           py37h74d4b33_0
[conda] pytorch                   1.10.0.dev20210729 py3.7_cuda11.1_cudnn8.0.5_0    pytorch-nightly
[conda] torch2trt                 0.2.0                     dev_0    <develop>
[conda] torchaudio                0.10.0.dev20210729            py37    pytorch-nightly
[conda] torchvision               0.11.0.dev20210729      py37_cu111    pytorch-nightly
```

## Additional context

Based on [ClassyVision commit 92b2255](https://github.com/facebookresearch/ClassyVision/commits/master)

```
 CUDNN_LOGDEST_DBG=stdout CUDNN_LOGINFO_DBG=1 python script.py | head -n 10

I! CuDNN (v8005) function cudnnCreate() called:
i!     handle: location=host; addr=0x557f3ac85a30;
i! Time: 2021-08-02T18:46:50.975435 (0d+0h+0m+4s since start)
i! Process=12389; Thread=12389; GPU=NULL; Handle=NULL; StreamId=NULL.


I! CuDNN (v8005) function cudnnSetStream() called:
i!     handle: type=cudnnHandle_t; streamId=(nil) (defaultStream);
i!     streamId: type=cudaStream_t; streamId=(nil) (defaultStream);
```

```
$ nvidia-smi
Mon Aug  2 18:50:10 2021
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 450.80.02    Driver Version: 450.80.02    CUDA Version: 11.0     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|                               |                      |               MIG M. |
|===============================+======================+======================|
|   0  Tesla V100-SXM2...  On   | 00000000:00:1B.0 Off |                    0 |
| N/A   33C    P0    36W / 300W |      0MiB / 16160MiB |      0%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
|   1  Tesla V100-SXM2...  On   | 00000000:00:1C.0 Off |                    0 |
| N/A   33C    P0    35W / 300W |      0MiB / 16160MiB |      0%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
|   2  Tesla V100-SXM2...  On   | 00000000:00:1D.0 Off |                    0 |
| N/A   33C    P0    39W / 300W |      0MiB / 16160MiB |      0%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
|   3  Tesla V100-SXM2...  On   | 00000000:00:1E.0 Off |                    0 |
| N/A   33C    P0    39W / 300W |      0MiB / 16160MiB |      0%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+

+-----------------------------------------------------------------------------+
| Processes:                                                                  |
|  GPU   GI   CI        PID   Type   Process name                  GPU Memory |
|        ID   ID                                                   Usage      |
|=============================================================================|
|  No running processes found                                                 |
+-----------------------------------------------------------------------------+
```

cc @fmassa @vfdev-5 @pmeier

**Comment by cpuhrsch** [comment]
Narrowed it down to BatchNorm2d

Script

```
import torch
from classy_vision.models import build_model

torch.manual_seed(123)
dtype = torch.float16
model = torch.nn.BatchNorm2d(64, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
model = model.eval().cuda().half()
model = model.to(dtype)
device = torch.device('cuda')
inp = torch.randn(2, 64, 50, 50, device=torch.device('cuda'), dtype=dtype)
out1 = model(inp)
inp = inp.contiguous(memory_format=torch.channels_last)
out2 = model(inp)

print((out1 - out2).abs().mean().item())
print(torch.allclose(out1, out2, rtol=1e-3, atol=1e-3))
```

Script output

```
1.12890625
False
```

**Comment by eqy** [comment]
I can repro this on `cuDNN` 8.2.0. I'll check to see if this is fixed in a later version.

**Comment by cpuhrsch** [comment]
@eqy - I found that this doesn't seem to call into cudnn to begin with. It [appears to fail the check](https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/Normalization.cpp#L436-L449), because the input is float16.

**Comment by eqy** [comment]
Nice catch! I can confirm that there doesn't seem to be any cuDNN kernels called here, just TensorIterator kernels. I'll see if I can find what's happening there.

**Comment by eqy** [comment]
Update: I think the issue is that the strides of the output tensor are being set to a channels first format when the underlying data layout is still in channels last. I'm not sure if this is a failure on the part of the `batch_norm_transform_input_channels_last_kernel` implementation (e.g., it does not do a layout transformation but is it actually supposed to?) or on the part of some other logic that is meant to propagate the memory format of the input tensor to the output tensor.

**Comment by cpuhrsch** [comment]
@eqy - makes sense. What do you think we should do next?

**Comment by eqy** [comment]
I think it might be a one-liner fix in `Normalization.cu`. Will open a PR if that works.
